src/bundle/Resources/public/js/scripts/admin.location.bookmark.js

```diff
@@ -1,42 +1,65 @@
 
-    const updateBookmarkLocationInput = doc.querySelector(SELECTOR_BOOKMARK_LOCATION_INPUT);
-    const currentLocationId = parseInt(updateBookmarkLocationInput.value, 10);
-
-    const submitBookmarkForm = () => doc.querySelector(SELECTOR_FORM).submit();
-    const updateBookmarkCheckbox = (bookmarked) => {
-        const checkbox = doc.querySelector(SELECTOR_BOOKMARK_CHECKBOX);
+        return response;
+    };
+    const handleRequestResponseStatus = (response) => {
+        return handleRequestError(response).status;
+    };
+    const handleUpdateError = global.eZ.helpers.notification.showErrorNotification;
+    const updateBookmark = (addBookmark) => {
+        const method = addBookmark ? 'POST' : 'DELETE';
+        const request = new Request(`${ENDPOINT_BOOKMARK}/${currentLocationId}`, {
+            method,
+            headers: {
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
 
-        checkbox.checked = bookmarked;
+        fetch(request)
+            .then(handleRequestResponseStatus)
+            .then(toggleBookmarkIconState.bind(null, addBookmark))
+            .catch(handleUpdateError);
     };
     const isCurrentLocation = (locationId) => {
         return parseInt(locationId, 10) === currentLocationId;
     };
     const toggleBookmarkIconState = (isBookmarked) => {
-        const wrapper = doc.querySelector(SELECTOR_BOOKMARK_WRAPPER);
-
-        wrapper.classList.toggle(CLASS_BOOKMARK_CHECKED, isBookmarked);
+        bookmarkWrapper.classList.toggle(CLASS_BOOKMARK_CHECKED, isBookmarked);
     };
-    const updateBookmarkForm = (event) => {
-        const { bookmarked, locationId } = event.detail;
+    const updateBookmarkIconState = ({ detail }) => {
+        const { bookmarked, locationId } = detail;
 
         if (isCurrentLocation(locationId)) {
-            updateBookmarkCheckbox(bookmarked);
             toggleBookmarkIconState(bookmarked);
         }
     };
-    const updateBookmarkIconState = (event) => {
-        const checked = event.target.checked;
+    const checkIsBookmarked = () => {
+        if (!bookmarkWrapper) {
+            return null;
+        }
+
+        return bookmarkWrapper.classList.contains(CLASS_BOOKMARK_CHECKED);
+    };
+    const onBookmarkChange = () => {
+        const addBookmark = !checkIsBookmarked();
 
-        toggleBookmarkIconState(checked);
+        updateBookmark(addBookmark);
     };
 
-    doc.body.addEventListener('ez-bookmark-change', updateBookmarkForm, false);
-    doc.querySelector(SELECTOR_BOOKMARK_CHECKBOX).addEventListener('change', submitBookmarkForm, false);
-    doc.querySelector(SELECTOR_BOOKMARK_CHECKBOX).addEventListener('change', updateBookmarkIconState, false);
+    doc.body.addEventListener('ez-bookmark-change', updateBookmarkIconState, false);
+    bookmarkWrapper.addEventListener('click', onBookmarkChange, false);
```

Comment:
I think you should add event to the checkbox on `change` and you will not need to check if wrapper has class but you will have the information about checked or not in the event.

Reply:
OK, in the new approach there is no checkbox, sorry ;)